spec/spec_helper.rb

```diff
@@ -48,6 +48,7 @@ def serve_once(body)
     yield
   ensure
     server.shutdown
+    thread.join(5)
     thread.exit
```

Comment:
I assume it won't make sense to exit anymore since it would be a dead thread after calling `#join`?